api/src/registry.rs

```diff
@@ -50,93 +51,100 @@ impl<'a> ImagesService<'a> {
+                        OciDistributionError::AuthenticationFailure(err) => {
+                            RegistryError::AuthenticationFailure {
+                                image: image.to_string(),
+                                failure: err,
+                            }
+                        }
+                        OciDistributionError::ImageManifestNotFoundError(_) => {
+                            RegistryError::ImageNotFound {
+                                image: image.to_string(),
+                            }
+                        }
+                        err => RegistryError::UnexpectedError {
+                            image: image.to_string(),
+                            internal_message: err.to_string(),
+                        },
+                    });
                 }
-            })
-            .collect::<HashMap<Image, ImageInfo>>())
+            }
+        }
+
+        Ok(image_infos)
     }
 
     async fn resolve_image_info<'i>(
         config: &Config,
         image: &'i Image,
-    ) -> Result<(&'i Image, Option<ImageInfo>), (&'i Image, ImagesServiceError)> {
+    ) -> Result<(&'i Image, ImageInfo), (&'i Image, OciDistributionError)> {
         debug!("Resolve image manifest for {:?}", image);
 
-        let client = Self::create_client(config, image)
-            .await
-            .map_err(|err| (image, err))?;
-
-        let (image_name, tag) = (image.name().unwrap(), image.tag().unwrap());
-
-        let manifest = client
-            .get_manifest(&image_name, &tag)
+        let mut client = Client::new(ClientConfig {
+            platform_resolver: Some(Box::new(|entries| {
+                oci_distribution::client::current_platform_resolver(entries).or(
+                    // There are cases where current_platform_resolver fails, e.g. in tests on
+                    // MacOS. However it is not safe to assume the current platform that PREvant
+                    // runs on it the platform the backend (Docker or Kubernetes) runs on. For
+                    // example, it could be the case that clusters have multiple architectures
+                    // https://carlosedp.medium.com/building-a-hybrid-x86-64-and-arm-kubernetes-cluster-e7f94ff6e51d
+                    //
+                    // Thus, the first entry will be used when current_platform_resolver fails and
+                    // it is assumed that the information provided by the image config is equal on
+                    // each platform. If a port mapping or a volume definition is different for
+                    // different platforms, that would cripple into issues not just for PREvant but
+                    // rather for all users that migrate to a different architecture.
+                    entries.first().map(|e| e.digest.clone()),
+                )
+            })),
+            ..Default::default()
+        });
+
+        let reference = Reference::from_str(&image.to_string()).unwrap();
```

Comment:
I think it would be safe to handle the error that Reference::from_str produces. 
I passed image as postgres_ and private-registry.example.com/_/postgres and got the following error:
thread 'rocket-worker-thread' panicked at src/registry.rs:121:65:
called `Result::unwrap()` on an `Err` value: ReferenceInvalidFormat

Problem is that once this has panicked, the tile goes in conflict and gets hard to recover.

Reply:
Nice catch. I included an additional check in `image.rs` which make sure you never reach that point.